tiledb/api/c_api/enumeration/test/unit_capi_enumeration.cc

```diff
@@ -0,0 +1,430 @@
+        5 * sizeof(uint64_t),
+        &enumeration_);
+    REQUIRE(rc == TILEDB_OK);
+  }
+
+  ~VarSizeEnumeration() {
+    tiledb_enumeration_free(&enumeration_);
+  }
+
+  ordinary_context ctx_;
+  tiledb_enumeration_t* enumeration_;
+};
+
+TEST_CASE(
+    "C API: tiledb_enumeration_alloc argument validation",
+    "[capi][enumeration]") {
+  ordinary_context ctx{};
+  tiledb_enumeration_t* enumeration;
+
+  int32_t values[5] = {1, 2, 3, 4, 5};
+  const char* data = "foobarbazbingobango";
+  uint64_t offsets[5] = {0, 3, 6, 9, 14};
+
+  SECTION("success - fixed size") {
+    auto rc = tiledb_enumeration_alloc(
+        ctx.context,
+        TILEDB_UINT32,
+        1,
+        0,
+        values,
+        sizeof(uint32_t) * 5,
+        nullptr,
+        0,
+        &enumeration);
+    REQUIRE(rc == TILEDB_OK);
+    tiledb_enumeration_free(&enumeration);
+  }
+
+  SECTION("success - var size") {
+    auto rc = tiledb_enumeration_alloc(
+        ctx.context,
+        TILEDB_STRING_ASCII,
+        UINT32_MAX,
+        0,
+        (void*)data,
+        strlen(data),
+        offsets,
+        sizeof(uint64_t) * 5,
+        &enumeration);
+    REQUIRE(rc == TILEDB_OK);
+    tiledb_enumeration_free(&enumeration);
+  }
+
+  SECTION("failure - null context") {
+    auto rc = tiledb_enumeration_alloc(
+        nullptr,
+        TILEDB_UINT32,
+        1,
+        0,
+        values,
+        sizeof(uint32_t) * 5,
+        nullptr,
+        0,
+        &enumeration);
+    REQUIRE(rc == TILEDB_INVALID_CONTEXT);
+    tiledb_enumeration_free(&enumeration);
+  }
+
+  SECTION("failure - invalid datatype") {
+    auto rc = tiledb_enumeration_alloc(
+        ctx.context,
+        (tiledb_datatype_t)255,
+        1,
+        0,
+        values,
+        sizeof(uint32_t) * 5,
+        nullptr,
+        0,
+        &enumeration);
+    REQUIRE(rc == TILEDB_ERR);
+    tiledb_enumeration_free(&enumeration);
+  }
+
+  SECTION("failure - data nullptr") {
+    auto rc = tiledb_enumeration_alloc(
+        ctx.context,
+        TILEDB_INT32,
+        1,
+        0,
+        nullptr,
+        sizeof(uint32_t) * 5,
+        nullptr,
+        0,
+        &enumeration);
+    REQUIRE(rc == TILEDB_ERR);
+    tiledb_enumeration_free(&enumeration);
+  }
+
+  SECTION("failure - data_size == 0") {
+    auto rc = tiledb_enumeration_alloc(
+        ctx.context, TILEDB_INT32, 1, 0, values, 0, nullptr, 0, &enumeration);
+    REQUIRE(rc == TILEDB_ERR);
+    tiledb_enumeration_free(&enumeration);
+  }
+
+  SECTION("failure - enumeration nullptr") {
+    auto rc = tiledb_enumeration_alloc(
+        ctx.context,
+        TILEDB_INT32,
+        1,
+        0,
+        values,
+        sizeof(uint32_t) * 5,
+        nullptr,
+        0,
+        nullptr);
+    REQUIRE(rc == TILEDB_ERR);
+    tiledb_enumeration_free(&enumeration);
+  }
+
+  SECTION("failure - offsets nullptr") {
+    auto rc = tiledb_enumeration_alloc(
+        ctx.context,
+        TILEDB_STRING_ASCII,
+        UINT32_MAX,
```

Comment:
Just did this because I remembered this comment. However it turned out pulling TILEDB_VAR_NUM from tiledb.h 
 causes a link dependency since TILEDB_VAR_NUM is a function under the macro. I opted for just doing this instead:

```c++
#ifndef TILEDB_VAR_NUM
#define TILEDB_VAR_NUM tiledb::sm::constants::var_num
#endif
```